### ISSUE TITLE
fakeifying tensor subclasses that are views is broken

### ISSUE THREAD

**Issue by bdhirsh** [pytorch/pytorch#115782]
Context: Int4 subclass ran into this here: https://github.com/pytorch-labs/ao/pull/13/files#r1425980422

min repro:
```
import torch
from torch.testing._internal.two_tensor import TwoTensor

@torch.compile
def f(x):
    return x * x
x = TwoTensor(torch.ones(2), torch.ones(2))
x_view = x.view(x.shape)
out = f(x_view)
```

Fails this assertion: https://github.com/pytorch/pytorch/blob/main/torch/_subclasses/meta_utils.py#L474

It looks like that's because:

(1) When we fakeify a tensor that has a `._base`, we need to fakeify the ._base

(2) When we start to fakeify the ._base, we unconditionally give it's base a `StatelessSymbolicContext` ([code](https://github.com/pytorch/pytorch/blob/43efe39cb12cb59ecd8df955f1a9dc8f47acb838/torch/_subclasses/meta_utils.py#L330))

(3) That's wrong - the base can itself be another subclass, in which case we should give it a `SubclassSymbolicContext`

cc @Chillee @ezyang @zou3519 @albanD @samdow @msaroufim @wconstab @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

**Comment by bdhirsh** [comment]
cc @jbschlosser, I wonder if your work around handling fakifying subclass views will end up fixing this issue

**Comment by voznesenskym** [comment]
This is known and discussed offline - I think we have a plan here.

**Comment by jbschlosser** [comment]
Closing as fixed in #118405.
